File: Demo.API/Model/Models/TicketModel.cs
using Model.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
namespace Model.Models
{
    public class TicketModel
    {

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int TicketID { get; set; }

        public TicketTypeEnum Type { get; set; }

        public SeverityEnum Severity { get; set; }

        public string Summary { get; set; }

        public string Description { get; set; }

        public bool IsResolved { get; set; }

        public bool IsDel { get; set; }

        public DateTime? ResolvedTime { get; set; }

        public string CreateBy { get; set; }

        public DateTime CreateTime { get; set; }
    }
}

File: DemoFrontend/src/api/ticket/model/ticketModel.ts
import { BasicFetchResult, PaginationParams } from '../../model/baseModel';

export type GetTicketModel = BasicFetchResult<boolean>;
export type GetTicketAllModel = BasicFetchResult<Ticket[]>;

export type OptionsModel = BasicFetchResult<Options[]>;

export interface GetTicketAllParams extends PaginationParams {
  types?: string;
  severities?: string;
  isResolved?: boolean;
  startTime?: string;
  endTime?: string;
}

export interface Ticket {
  ticketID: number;
  type: number;
  severity: number;
  summary: string;

  description: string;
  isResolved: boolean;
  isDel: boolean;
  resolvedTime?: number;
  createBy: string;
  createTime: number;
}

export interface Options {
  name: string;
  value: number;
}
